addons/hr_expense/models/res_config_settings.py

```diff
@@ -28,7 +28,8 @@ def set_values(self):
         super(ResConfigSettings, self).set_values()
         self.env.ref('hr_expense.mail_alias_expense').write({'alias_name': self.expense_alias_prefix})
 
-    @api.onchange('use_mailgateway')
-    def _onchange_use_mailgateway(self):
-        if not self.use_mailgateway:
-            self.expense_alias_prefix = False
+    @api.depends('use_mailgateway')
+    def _compute_expense_alias_prefix(self):
+        for settings in self:
+            if not settings.use_mailgateway:
+                settings.expense_alias_prefix = False
```

Comment:
Yep, but we want to avoid using them, except in the rare cases for which it's still impossible. 

```python3
self.filtered(lambda w: not w.use_mailgateway).update({'expense_alias_prefix': False})
```